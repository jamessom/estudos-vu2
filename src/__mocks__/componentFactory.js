import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from '@/router/index';

export default function componentFactory(options = {}) {
  return {
    wrapper: () => {
      const { component, componentOptions, withRouter } = options;
      const localvue = createLocalVue();

      if (withRouter) {
        localvue.use(VueRouter);

        return shallowMount(component, {
          localvue,
          ...componentOptions,
        });
      }

      return shallowMount(component, {
        localvue,
        ...componentOptions,
      });
    },
  };
}
