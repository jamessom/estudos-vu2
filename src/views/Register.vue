<template>
  <section class="section">
    <div class="columns">

      <div class="column">
        <h1>Pré-cadastro de Torcedor</h1>

        <input-base label="Nome completo" :placeholder="this.tips.fullName" />
        <gender-select label="Sexo" :options=genders />
        <input-base label='Email' type="email" :placeholder="this.tips.email" />
        <input-base label='Repetir e-mail' type="email" :placeholder="this.tips.reEmail" />
        <gender-select label="Tipo de documento" :options=documentType />
        <input-base label='Número do documento' :placeholder="this.tips.docNumber" />
        <input-base label='Senha' type='password' :placeholder="this.tips.password" />
        <input-base label='Repetir senha' type='password' :placeholder="this.tips.rePassword" />
      </div>
    </div>
  </section>
</template>

<script>
import InputBase from '@/components/InputBase.vue';
import GenderSelect from '@/components/GenderSelect.vue';

export default {
  name: 'Register',
  data() {
    return {
      tips: {
        fullName: 'Por favor informe seu nome completo',
        email: 'Por favor informe seu email',
        reEmail: 'Por faver informe seu email novamente',
        docNumber: 'Por favor informe o número do documento selecionado',
        password: 'Por favor informe uma senha',
        rePassword: 'Por favor informe sua senha novamente',
      },
      genders: [
        { value: 'selected', text: 'Selecione uma opção' },
        { value: 'M', text: 'Masculino' },
        { value: 'F', text: 'Feminino' },
        { value: 'O', text: 'Outros' },
      ],
      documentType: [
        { value: 'selected', text: 'Selecione uma opção' },
        { value: 1, text: 'RG' },
        { value: 2, text: 'CPF' },
        { value: 3, text: 'CNH' },
        { value: 4, text: 'Certidão de Nascimento' },
        { value: 5, text: 'Passaporte' },
      ],
    };
  },
  components: {
    InputBase,
    GenderSelect,
  },
};
</script>
