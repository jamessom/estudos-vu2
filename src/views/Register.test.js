import componentFactory from '@/__mocks__/componentFactory';
import Register from './Register.vue';

describe('src/views/Register.vue', () => {
  const factory = componentFactory({
    component: Register,
  });

  it('should renders correctly', () => {
    expect(factory.wrapper().element).toMatchSnapshot();
  });
});
