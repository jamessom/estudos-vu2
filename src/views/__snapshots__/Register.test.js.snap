// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/views/Register.vue should renders correctly 1`] = `
<section
  class="section"
>
  <div
    class="columns"
  >
    <div
      class="column"
    >
      <h1>
        Pré-cadastro de Torcedor
      </h1>
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="text"
        label="Nome completo"
        placeholder="Por favor informe seu nome completo"
      />
       
      <gender-select-stub
        label="Sexo"
        options="[object Object],[object Object],[object Object],[object Object]"
      />
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="text"
        label="Email"
        placeholder="Por favor informe seu email"
        type="email"
      />
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="text"
        label="Repetir e-mail"
        placeholder="Por faver informe seu email novamente"
        type="email"
      />
       
      <gender-select-stub
        label="Tipo de documento"
        options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      />
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="text"
        label="Número do documento"
        placeholder="Por favor informe o número do documento selecionado"
      />
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="text"
        label="Senha"
        placeholder="Por favor informe uma senha"
        type="password"
      />
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="text"
        label="Repetir senha"
        placeholder="Por favor informe sua senha novamente"
        type="password"
      />
    </div>
  </div>
</section>
`;
